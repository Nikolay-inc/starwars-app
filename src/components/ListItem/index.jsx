import {
  AccordionItem,
  AccordionButton,
  AccordionPanel,
  AccordionIcon,
  Box,
  Stack,
  StackDivider,
  Text,
} from '@chakra-ui/react';
import { Link } from '@chakra-ui/next-js';
import { urlParser } from '../../helpers/urlParser';

// list item for heroes
const ListItem = ({ person }) => {
  return (
    <AccordionItem mt='15px'>
      <h2>
        <AccordionButton>
          <Box as='span' flex='1' textAlign='left'>
            <Link 
              href={`/people/${urlParser(person.url)}`} 
              fontWeight='lite'
              fontSize='20px'
            >
              {person.name}
            </Link>
          </Box>
          <AccordionIcon />
        </AccordionButton>
      </h2>
      <AccordionPanel pb={4}>
      <div>
        {/* StackDivider automatically divides boxes */}
        <Stack divider={<StackDivider />} spacing='4'>
          <Box>
            <Text pt='2'>
              Height - {person.height}
            </Text>
            <Text pt='2'>
              Mass - {person.mass}
            </Text>
          </Box>
          <Box>
            <Text pt='2'>
              Hair Color - {person.hair_color}
            </Text>
            <Text pt='2'>
              Skin Color - {person.skin_color}
            </Text>
            <Text pt='2'>
              Eye Color - {person.eye_color}
            </Text>
          </Box>
          <Box>
            <Text pt='2'>
              Birth Year - {person.birth_year}
            </Text>
            <Text pt='2'>
              Gender - {person.gender}
            </Text>
          </Box>
        </Stack>
      </div>
      </AccordionPanel>
    </AccordionItem>
  );
}

export default ListItem;