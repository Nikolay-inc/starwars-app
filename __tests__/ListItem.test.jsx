import { render, screen } from '@testing-library/react';
import ListItem from '../src/components/ListItem';
import { Accordion } from '@chakra-ui/react';

describe('ListItem component', () => {
  // mock person data
  const person = {
    name: 'Luke Skywalker',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    url: 'https://sw-api.starnavi.io/people/1/',
  };

  test('renders person name as a link', () => {
    render(
      <Accordion allowMultiple>
        <ListItem person={person} />
      </Accordion>
    );
    const linkElement = screen.getByRole('link', { name: /Luke Skywalker/i });
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute('href', '/people/1');
  });

  test('renders person details', () => {
    render(
      <Accordion allowMultiple>
        <ListItem person={person} />
      </Accordion>
    );
    expect(screen.getByText(/Height - 172/i)).toBeInTheDocument();
    expect(screen.getByText(/Mass - 77/i)).toBeInTheDocument();
    expect(screen.getByText(/Hair Color - blond/i)).toBeInTheDocument();
    expect(screen.getByText(/Skin Color - fair/i)).toBeInTheDocument();
    expect(screen.getByText(/Eye Color - blue/i)).toBeInTheDocument();
    expect(screen.getByText(/Birth Year - 19BBY/i)).toBeInTheDocument();
    expect(screen.getByText(/Gender - male/i)).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(
      <Accordion allowMultiple>
        <ListItem person={person} />
      </Accordion>
    );
    expect(container).toMatchSnapshot();
  });
});
