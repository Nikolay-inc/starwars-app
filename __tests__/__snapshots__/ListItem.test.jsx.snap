// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component matches snapshot 1`] = `
<div>
  <div
    class="chakra-accordion css-0"
  >
    <div
      class="chakra-accordion__item css-1te8kxg"
    >
      <h2>
        <button
          aria-controls="accordion-panel-:r4:"
          aria-expanded="false"
          class="chakra-accordion__button css-ubg1nw"
          data-index="0"
          id="accordion-button-:r4:"
          type="button"
        >
          <span
            class="css-1eziwv"
          >
            <a
              class="chakra-link css-jbu2m3"
              href="/people/1"
            >
              Luke Skywalker
            </a>
          </span>
          <svg
            aria-hidden="true"
            class="chakra-icon chakra-accordion__icon css-8o5ky2"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h2>
      <div
        class="chakra-collapse"
        style="overflow: hidden; display: none; opacity: 0; height: 0px;"
      >
        <div
          aria-labelledby="accordion-button-:r4:"
          class="chakra-accordion__panel css-b0qdn7"
          id="accordion-panel-:r4:"
          role="region"
        >
          <div>
            <div
              class="chakra-stack css-j7qwjs"
            >
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-1tqud6q"
                >
                  Height - 
                  172
                </p>
                <p
                  class="chakra-text css-1tqud6q"
                >
                  Mass - 
                  77
                </p>
              </div>
              <div
                class="chakra-stack__divider css-1jj487w"
              />
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-1tqud6q"
                >
                  Hair Color - 
                  blond
                </p>
                <p
                  class="chakra-text css-1tqud6q"
                >
                  Skin Color - 
                  fair
                </p>
                <p
                  class="chakra-text css-1tqud6q"
                >
                  Eye Color - 
                  blue
                </p>
              </div>
              <div
                class="chakra-stack__divider css-1jj487w"
              />
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-1tqud6q"
                >
                  Birth Year - 
                  19BBY
                </p>
                <p
                  class="chakra-text css-1tqud6q"
                >
                  Gender - 
                  male
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
